pints/_mcmc/_differential_evolution.py

```diff
@@ -268,7 +268,13 @@ def set_hyper_parameters(self, x):
         """
         self.set_gamma(x[0])
         self.set_scale_coefficient(x[1])
-        self.set_gamma_switch_rate(x[2])
+        try:
+            int_x2 = int(x[2])
+        except ValueError:
```

Comment:
I think it can be a TypeError too, so perhaps `except (ValueError, TypeError)`
(Alternative is not too catch at all, which is generally ok, but this is nicer of course!)